### DISCUSSION TITLE
CSS and AppTheme

### DISCUSSION THREAD

**Discussion by tomeverin** [xamarin/Xamarin.Forms#11403] [General]
My app is themed using embedded CSS with typical XAML like:

```
<ContentPage.Resources>
    <StyleSheet Source="/Embedded/mystylesheet.css" />
</ContentPage.Resources>
```

4.6.x added AppTheme functionality (https://docs.microsoft.com/en-us/xamarin/xamarin-forms/release-notes/4.6/4.6.0) and indicates the following should work:

```
<StyleSheet>
    <StyleSheetExtension.Style>
	<OnAppTheme x:TypeArguments="x:String">
	    <OnAppTheme.Light>
		<![CDATA[
		    #cssStyledLabel {
			color: purple;
		    }
		]]>
	    </OnAppTheme.Light>
	    <OnAppTheme.Dark>
		<![CDATA[
		    #cssStyledLabel {
			color: orange;
		    }
		]]>
	    </OnAppTheme.Dark>
	</OnAppTheme>
    </StyleSheetExtension.Style>
</StyleSheet>
```

Is there currently a way that I can implement app themes as follows?

```
<ContentPage.Resources>
    <StyleSheet>
        <StyleSheetExtension.Style>
	    <OnAppTheme>
                 <OnAppTheme.Light>
                     <StyleSheet Source="/Embedded/mystylesheet-light.css" />
                 </OnAppTheme.Light>
                 <OnAppTheme.Dark>
                     <StyleSheet Source="/Embedded/mystylesheet-dark.css" />
                 </OnAppTheme.Dark>
        </StyleSheetExtension.Style>
    </StyleSheet>
</ContentPage.Resources>
```

I can't get any of the AppTheme CSS stuff to work. I've added the experimental flag and turned off XAMLC. With the XAML with the CDATA styles above: none of the styles are applied.

Any guidance on the latest approach?

**Comment by davidortinau** [comment]
In order to make it work at runtime for expected scenarios, the feature was refactored in preview. As a result, the 4.6 guidance is outdated. This doesn't often happen, but it's the reason we mark these as "experimental" in order to work out the issues.

@StephaneDelcroix is looking into what we can do to make this work with CSS.

**Comment by tomeverin** [comment]
Thanks @davidortinau 

Cheers for looking into it @StephaneDelcroix , if you have any estimates on how likely this support is I'd be happy to hear about it. It looks like my best option is to convert all my CSS into xaml styles.
